brain/compartmentReportReader.cpp

```diff
@@ -0,0 +1,203 @@
+}
+
+CompartmentReportFrame::CompartmentReportFrame()
+    : _impl{new detail::CompartmentReportFrame}
+{
+}
+
+CompartmentReportFrame::~CompartmentReportFrame()
+{
+}
+
+CompartmentReportFrame::CompartmentReportFrame(
+    CompartmentReportFrame&& other) noexcept : _impl(std::move(other._impl))
+{
+}
+
+CompartmentReportFrame& CompartmentReportFrame::operator=(
+    CompartmentReportFrame&& other) noexcept
+{
+    if (&other == this)
+        return *this;
+    _impl = std::move(other._impl);
+    return *this;
+}
+
+bool CompartmentReportFrame::empty() const
+{
+    return _impl->data.empty();
+}
+
+const brion::floats& CompartmentReportFrame::getData() const
+{
+    return _impl->data;
+}
+
+float CompartmentReportFrame::getTimestamp() const
+{
+    return _impl->timeStamp;
+}
+
+CompartmentReportView::CompartmentReportView(
+    const std::shared_ptr<detail::CompartmentReportReader>& readerImpl,
+    const brion::GIDSet& gids)
+    : _impl(new detail::CompartmentReportView(readerImpl, gids))
+{
+}
+
+CompartmentReportView::CompartmentReportView(CompartmentReportView&& other)
+{
+    _impl = std::move(other._impl);
+}
+
+CompartmentReportView& CompartmentReportView::operator=(
+    CompartmentReportView&& other)
+{
+    _impl = std::move(other._impl);
+    return *this;
+}
+
+CompartmentReportView::~CompartmentReportView()
+{
+}
+
+const brion::GIDSet& CompartmentReportView::getGIDs() const
+{
+    return _impl->report->getGIDs();
+}
+
+const CompartmentReportMapping& CompartmentReportView::getMapping() const
+{
+    return _impl->mapping;
+}
+
+std::future<CompartmentReportFrame> CompartmentReportView::load(float timestamp)
+{
+    auto report = _impl->report;
+    auto loadFrameTask = [timestamp, report] {
+
+        CompartmentReportFrame frame;
+        frame._impl->timeStamp = timestamp;
+
+        auto data = report->loadFrame(timestamp);
+        if (data)
+            frame._impl->data = std::move(*data);
+
+        return frame;
+    };
+
+    return _impl->readerImpl->threadPool.post(loadFrameTask);
+}
+
+std::future<CompartmentReportFrames> CompartmentReportView::load(float start,
+                                                                 float end)
+{
+    start = std::max(start, _impl->report->getStartTime());
+    end = std::min(end, _impl->report->getEndTime());
+
+    if (end <= start)
+        throw std::logic_error("Invalid interval");
+
+    float timestep = _impl->report->getTimestep();
+
+    auto report = _impl->report;
+
+    auto loadFrameTask = [start, end, timestep, report] {
+
+        CompartmentReportFrames frames;
+        size_t i = 0;
+        for (float t = start; t <= end; t += timestep, ++i)
+        {
+            CompartmentReportFrame frame;
+            frame._impl->timeStamp = t;
+            auto data = report->loadFrame(t);
+            if (data)
+                frame._impl->data = std::move(*data);
+
+            frames.push_back(std::move(frame));
```

Comment:
No. emplace_back just takes the constructor arguments.
Move is always explicit.